src/main.rs

```diff
@@ -1,258 +1,13 @@
 extern crate structopt;
 extern crate xcb;
-mod util;
+mod lib;
 
+use lib::parse_args::Opt;
+use lib::{
+    fill_format_string, get_window_at_point, get_window_geom, grab_pointer_set_cursor, set_shape,
+    set_title, HacksawResult, CURSOR_GRAB_TRIES,
+};
```

Comment:
Minor nit, but I'm personally not a fan of importing so many individual members of a module.

Reply:
What would you prefer?

Reply:
One thing I also want to do is move parts of run() into functions - this might reduce imports

Reply:
I'd replace usage with `lib::...`.

Reply:
> One thing I also want to do is move parts of run() into functions - this might reduce imports

Speaking of which, maybe we should implement the [`Termination` trait](https://doc.rust-lang.org/edition-guide/rust-2018/error-handling-and-panics/question-mark-in-main-and-tests.html) instead of wrapping main the way I did.

Reply:
As it is, we could just return Result (since right now we exit 0 / 1 and Result will exit EXIT_SUCCESS or EXIT_FAILURE)

Reply:
I think the alternative would be to newtype an integral type and impl Termination for that.

Reply:
https://github.com/neXromancers/hacksaw/commit/ad17a5a4238095a9ea51bc293ed10145b7bbfda1